search/search.go

```diff
@@ -279,6 +279,7 @@ type SearcherOptions struct {
 type SearchContext struct {
 	DocumentMatchPool *DocumentMatchPool
 	Collector         Collector
+	IndexReader       index.IndexReader
```

Comment:
wondering why this is needed?  it seems like the method you refactored passes the IndexReader as an argument, and I don't see search context used elsewhere in this change

Reply:
The thinking is that, the "MakeDocumentMatchHandler"  ref: "https://github.com/blevesearch/bleve/blob/master/search/collector.go#L51 here takes the SearchContext as an argument. And when the application gets this builder callback, then it can use this SearchContext information to invoke these LoadFieldsAndHighlight()/similar methods in bleve.

Reply:
For eg: the Collector inside the context can be used by the application to get a Collector chaining effect. Like, the application can process the hit on https://github.com/blevesearch/bleve/blob/master/search/collector.go#L40 callback and later pass it back to the topN collector handler like here: https://github.com/blevesearch/bleve/blob/master/search/collector/topn_test.go#L588

Reply:
So, I'm not completely sure I follow your scenario yet, but my understanding from the commit message was that this was just a refactor of one really long method (SearchInContext) into two slightly less long methods.  That was an obvious good improvement.  However, this second change seems unrelated (unless I still haven't gotten the connection).

The reason it seemed confusing to me was that generally, everyone that needs the IndexReader, is directly passed a copy of it today.  The SearchContext was added later, and my thinking was that everyone that got the SearchContext already got the IndexReader (usually as a separate argument).

Regarding your scenario you describe, are you suggesting that loading stored fields and highlighting should be done inside a collector (or collector chain) and not done outside it?